core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -278,6 +294,245 @@ impl SanitizedTransactionReceiveAndBuffer {
     }
 }
 
+pub(crate) struct TransactionViewReceiveAndBuffer {
+    pub receiver: BankingPacketReceiver,
+    pub bank_forks: Arc<RwLock<BankForks>>,
+}
+
+impl ReceiveAndBuffer for TransactionViewReceiveAndBuffer {
+    type Transaction = RuntimeTransaction<ResolvedTransactionView<SharedBytes>>;
+    type Container = TransactionViewStateContainer;
+
+    fn receive_and_buffer_packets(
+        &mut self,
+        container: &mut Self::Container,
+        timing_metrics: &mut SchedulerTimingMetrics,
+        count_metrics: &mut SchedulerCountMetrics,
+        decision: &BufferedPacketsDecision,
+    ) -> bool {
+        let (root_bank, working_bank) = {
+            let bank_forks = self.bank_forks.read().unwrap();
+            let root_bank = bank_forks.root_bank();
+            let working_bank = bank_forks.working_bank();
+            (root_bank, working_bank)
+        };
+
+        // Receive packet batches.
+        const TIMEOUT: Duration = Duration::from_millis(10);
```

Comment:
I'm thinking this will need to be profiled and tweaked. E.g. pull for a certain amount of time or up to some packet limit to make sure we don't spend all our time receiving and cleaning (and thus less time on actually scheduling to feed the workers)

Reply:
Yeah. Want to make these more flexible instead of just constants.

After this PR lands, I have #4064 to hoist all the prio-graph constants into a config.
I am hoping to do something similar for the controller's timeouts.